ttnn/cpp/ttnn/operations/matmul/device/matmul_op_multi_core_reuse_mcast_dram_sharded_program_factory.cpp

```diff
@@ -647,9 +647,7 @@ tt::tt_metal::operation::ProgramWithCallbacks create_program_dram_sharded(
     }
 
     std::vector<CoreCoord> mcast_receiver_coords = corerange_to_cores(mcast_receivers);
-    for (uint32_t i = 0; i < mcast_receiver_coords.size(); ++i) {
-        auto core = mcast_receiver_coords[i];
-
+    for (const auto& core : mcast_receiver_coords) {
```

Comment:
```suggestion
    for (auto core : mcast_receiver_coords) {
```

Align with original code to avoid thinking.